js/runtime.js

```diff
@@ -796,6 +757,59 @@
                     ctx.ellipse(pt.x, pt.y, rad1, rad2, rot, 0, Math.PI * 2);
                 });
             },
+            polygon: function(args){
+                var points = [].slice.call(arguments);
+                return new util.Shape(function(ctx){
+                    ctx.beginPath();
+                    var points = this.pointsArray;
+                    for(var i=0; i < points.length; i++) {
+                        if (i === 0) {
+                            ctx.moveTo(points[i].x, points[i].y);
+                        }
+                        else {
+                            ctx.lineTo(points[i].x, points[i].y);
+                        }
+                        if(i === points.length-1) ctx.lineTo(points[0].x, points[0].y);
+                    }
+                }, points);
+            },
+            lineTo: function(startPoint, toPoint){
+                return new util.Path(getContext().lineTo, new Array(toPoint.x, toPoint.y), startPoint);
+            },
+            arc: function(radius, startPoint, controlPoint1, controlPoint2){
+                var path = new util.Path(getContext().arcTo, new Array(controlPoint1.x,
+                                                            controlPoint1.y,controlPoint2.x, controlPoint2.y,
+                                                            radius), startPoint);
+                return path;
+            },
+            bezierCurve: function(startPoint, toPoint, controlPoint1, controlPoint2){
+                var path = new util.Path(getContext().bezierCurveTo, new Array(controlPoint1.x, controlPoint1.y,
+                                                                    controlPoint2.x, controlPoint2.y, toPoint.x,
+                                                                    toPoint.y), startPoint);
+                return path;
+            },
+            quadraticCurve: function(startPoint, toPoint, controlPoint){
+                var path = new util.Path(getContext().quadraticCurveTo, new Array(controlPoint.x,
+                                                                       controlPoint.y,toPoint.x, toPoint.y),
+                                                                       startPoint);
+                return path;
+            },
+            path: function(args){
+                var paths = [].slice.call(arguments);
+                return new util.Shape(paths);
+            },
+            closePath: function(){
+                return new util.Path(getContext().closePath);
+            },
+            lastPoint: function(){
+                return false;
+            },
+            lineStyle: function(width, color, capStyle, joinStyle){
```

Comment:
We have `lineStyle` under colour too (that version only sets the color). I think we should have `lineColor`, `lineWidth` (we have that), `lineCapStyle`, and `lineJoinStyle` as separate steps and retire the one that does all four at once.